cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -23,28 +23,48 @@
 from charms.templating.jinja2 import render
 
 
-def _reconfigure_docker_for_sdn():
-    ''' By default docker uses the docker0 bridge for container networking.
-    This method removes the default docker bridge, and reconfigures the
-    DOCKER_OPTS to use the flannel networking bridge '''
+@hook('upgrade-charm')
+def remove_installed_state():
+    remove_state('kubernetes.worker.bins.installed')
 
-    hookenv.status_set('maintenance', 'Reconfiguring docker network bridge')
-    host.service_stop('docker')
-    apt_install(['bridge-utils'], fatal=True)
-    # cmd = "ifconfig docker0 down"
-    # ifconfig doesn't always work. use native linux networking commands to
-    # mark the bridge as inactive.
-    cmd = ['ip', 'link', 'set', 'docker0', 'down']
-    check_call(cmd)
 
-    cmd = ['brctl', 'delbr', 'docker0']
-    check_call(cmd)
-    set_state('docker.restart')
+@when('kubernetes.worker.bins.installed')
+@when_not('kube-dns.available')
+def notify_user_transient_status():
+    ''' During deployment the worker has to start kubelet without cluster dns
+    configured. If this is the first unit online in a service pool its waiting
+    to self host the dns pod, and configure itself to query the dns service
+    declared in the kube-system namespace '''
+
+    # Daemon options are managed by the FlagManager class
+    kubelet_opts = FlagManager('kubelet')
 
+    # Query the FlagManager dict for the dns option, and determine if
+    # the service is running
+    if ('--cluster-dns' not in kubelet_opts.data and
+       _systemctl_is_active('kubelet')):
+        hookenv.status_set('waiting', 'Waiting on cluster dns')
+    else:
+        hookenv.status_set('waiting',
+                           'Waiting for cluster-manager to start')
 
-@hook('upgrade-charm')
-def remove_installed_state():
-    remove_state('kubernetes.worker.bins.installed')
+
+# Handle cases where the DNS/Addons still haven't propigated
+@when('kubernetes.worker.bins.installed', 'kube-dns.available')
+def notify_user_turnup_status(kube_dns):
+    ''' Determine which state we are in and report that we are pending an
+    pods to start during the initial phase of deployment. '''
+
+    # Daemon options are managed by the FlagManager class
+    kubelet_opts = FlagManager('kubelet')
+
+    # Query the FlagManager dict for the dns option, and determine if
+    # the service is running
+    if ('--cluster-dns' in kubelet_opts.data and
+       _systemctl_is_active('kubelet')):
+        hookenv.status_set('active', 'Kubernetes worker running.')
+    else:
+        hookenv.status_set('waiting', 'Waiting on cluster dns')
```

Comment:
We should change this status so we can tell the difference from the state of the same name above.